docs/secure-agent-communication.asciidoc

```diff
@@ -227,9 +227,37 @@ Elastic APM agents can send unauthenticated (anonymous) events to the APM Server
 An event is considered to be anonymous if no authentication token can be extracted from the incoming request.
 By default, these anonymous requests are rejected and an authentication error is returned.
 
-In some cases, however, it makes sense to allow anonymous requests --
-for example, when using an agent that runs on the client, like the Real User Monitoring (RUM) agent running in a browser,
-or the iOS/Swift agent running in a user application.
+In some cases, however, it makes sense to allow anonymous requests -- for
+example, when using the Real User Monitoring (RUM) agent running in a browser,
+or the iOS/Swift agent running in a user application,
+it is not possible to hide or protect a secret token or API key.
+Thus, enabling anonymous authentication is required to ingest client-side APM data.
 
-Because anyone can send anonymous events to the APM Server,
-additional configuration variables are available to rate limit the number anonymous events the APM Server processes.
+[float]
+[[anonymous-auth-config]]
+=== Configuring anonymous authentication
+
+There are a few configuration variables that can mitigate the impact of malicious requests to an
+unauthenticated APM Server endpoint.
+
+Use the **Allowed anonymous agents** and **Allowed anonymous services** configs to ensure that the
+`agent.name` and `service.name` of each incoming request match a specified list.
+
+Additionally, the APM Server can rate-limit unauthenticated requests based on the client IP address
+(`client.ip`) of the request.
+This allows you to specify the maximum number of requests allowed per unique IP address, per second.
+
+[float]
+[[derive-client-ip]]
+=== Deriving an incoming request's `client.ip` address
+
+Due to CDNs, proxies, etc., the remote IP address of an incoming request might be different
```

Comment:
I don't think CDNs are relevant to APM Server. I'd just say proxies or load balancers.

Reply:
```suggestion
Due to proxies, the remote IP address of an incoming request might be different
```